File: TestMVVM/app/src/main/java/com/example/testmvvm/entidades/Lugar.java
package com.example.testmvvm.entidades;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.example.testmvvm.constantes.Constantes;


@Entity(tableName = Constantes.NOMBRE_TABLA_POSICION)
public class Lugar {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    public int id;

    @ColumnInfo(name = "x")
    public int x;

    @ColumnInfo(name = "y")
    public int y;

    @ColumnInfo(name = "nombre")
    public String nombre;

    public Lugar(int x, int y, String nombre) {
        this.x = x;
        this.y = y;
        this.nombre = nombre;
        //this.id=0;
    }

    public int getXValue() {
        return this.x;
    }
    public int getId() {
        return this.id;
    }
    public int getYValue() {
        return this.y;
    }

    public String getNombreValue() {
        return this.nombre;
    }

    public Lugar() {}
}

File: RoomTestKotlin/app/src/main/java/com/example/roomtestkotlin/modelo/entidades/Lugar.kt
package com.example.roomtestkotlin.modelo.entidades

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.roomtestkotlin.constantes.Constantes

@Entity(tableName = Constantes.NOMBRE_TABLA_POSICION) /*@Entity(tableName = "rutas",
    //caso primary key compuesta
        PrimaryKeys{"id_lugar","id_aaa"},

    //  foreignKeys compuesta
        foreignKeys = {
                @ForeignKey(entity = Lugar.class,
                        parentColumns = "id_lugar",
                        childColumns = "origen",
                        OnDelete=CASCADE),                @ForeignKey(entity = Lugar.class,
                        parentColumns = "id_lugar",
                        childColumns = "destino")
        }
)*/
class Lugar {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    var id = 0

    @ColumnInfo(name = "x")
    var x = 0

    @ColumnInfo(name = "y")
    var y = 0

    @ColumnInfo(name = "nombre")
    var nombre: String? = null

    constructor(x: Int, y: Int, nombre: String?) {
        this.x = x
        this.y = y
        this.nombre = nombre
    }

    fun getXValue(): Int {
        return this.x
    }

    fun getYValue(): Int {
        return this.y
    }

    fun getNombreValue(): String? {
        return this.nombre
    }



    constructor() {}
}